shared/middleware/helpers/bucket_helper.rb

```diff
@@ -290,6 +290,31 @@ def delete_multiple(encrypted_channel_id, filenames)
     s3.delete_objects(bucket: @bucket, delete: {objects: objects, quiet: true})
   end
 
+  def hard_delete_channel_content(encrypted_channel_id)
+    # TODO: Handle pagination in the S3 APIs
+    owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
+    # Find all versions of all objects
+    channel_prefix = s3_path owner_id, channel_id
+    version_list = s3.list_object_versions(bucket: @bucket, prefix: channel_prefix)
+    return 0 if version_list.versions.empty? && version_list.delete_markers.empty?
```

Comment:
would be nice to include a method-level comment explaining that this represents the number of items deleted.